pkg/tdexdconnect/tdexdconnect.go

```diff
@@ -0,0 +1,73 @@
+package tdexdconnect
+
+import (
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+)
+
+// Encode encodes the given args as query parameters of the returned *url.URL.
+func Encode(
+	rpcServerAddr, network string, certBytes, macBytes []byte,
+) (*url.URL, error) {
+	u := url.URL{Scheme: "tdexdconnect", Host: rpcServerAddr}
+	q := u.Query()
+
+	block, _ := pem.Decode(certBytes)
+	if block == nil || block.Type != "CERTIFICATE" {
+		return nil, fmt.Errorf("failed to decode PEM block containing certificate")
+	}
+	certificate := base64.RawURLEncoding.EncodeToString(block.Bytes)
+	q.Add("cert", certificate)
+
+	if macBytes != nil {
+		macaroon := base64.RawURLEncoding.EncodeToString(macBytes)
+		q.Add("macaroon", macaroon)
+	}
+
+	q.Add("net", network)
```

Comment:
what's this about

Reply:
If we move the fragmenter features inside the daemon, the client (CLI or UI) do not need to handle any network check?